util.py

```diff
@@ -0,0 +1,208 @@
+        self.client = influxdb_client.InfluxDBClient(
+            url=url, org=self.org, username=self.username, password=self.password, token=self.token
+        )
+        self.write_api = self.client.write_api(write_options=SYNCHRONOUS)
+        self.query_api = self.client.query_api()
+
+    def close(self):
+        self.write_api.close()
+        self.client.close()
+
+    def get_last(self, sensor_type: SensorType) -> int | float:
+        with self._lock:
+            query = f'from(bucket:"{self.bucket}")\
+                |> range(start: 0)\
+                |> filter(fn:(r) => r._measurement == "{sensor_type.name}")\
+                |> last()'
+
+            try:
+                result = self.query_api.query(org=self.org, query=query)
+                return result[0].records[0].get_value()
+            except IndexError:
+                logging.info("%s table is empty", sensor_type.name)
+                return float('nan')
+            except NewConnectionError:
+                logging.exception("InfluxDB Connection error, couldn't write")
+                return float('nan')
+            except ApiException:
+                logging.exception("InfluxDB ApiException, couldn't write")
+                return float('nan')
+
+    def add(self, sensor_type: SensorType, value: int | float):
+        with self._lock:
+            point = influxdb_client.Point(sensor_type.name).field("value", value)
+            try:
+                self.write_api.write(bucket=self.bucket, org=self.org, record=point)
+            except NewConnectionError:
+                logging.exception("InfluxDB Connection error, couldn't write")
+            except ApiException:
+                logging.exception("InfluxDB ApiException, couldn't write")
+
+
+class SensorReadings:
+    def __init__(self, database: Database) -> None:
+        self.readings = {
+            SensorType.TEMPERATURE: database.get_last(SensorType.TEMPERATURE),
+            SensorType.HUMIDITY: database.get_last(SensorType.HUMIDITY),
+            SensorType.PRESSURE: database.get_last(SensorType.PRESSURE),
+            SensorType.PM1: database.get_last(SensorType.PM1),
+            SensorType.PM2_5: database.get_last(SensorType.PM2_5),
+            SensorType.PM10: database.get_last(SensorType.PM10),
+        }
+        self.database = database
+
+    def get(self, sensor_type: SensorType):
+        return self.readings[sensor_type]
+
+    def add(self, sensor_type: SensorType, value: int | float):
+        self.readings[sensor_type] = value
+        self.database.add(sensor_type, value)
+
+class ResettableTimer(Timer):
+    """call start to initialize, call reset to 'start' timer"""
+    def __init__(self, interval: float, function: Callable[..., object], *args, **kwargs) -> None:
+        super().__init__(interval, function, args, kwargs)
+        self._function_wait = Condition()
+        self._end = False
+
+    def cancel(self):
```

Comment:
`-> ?`
We have type hints in some places, in some not, can't we align convention to have an equal one?